jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/Jackson2Annotator.java

```diff
@@ -129,9 +136,13 @@ public void objectBuilder(JDefinedClass clazz, JDefinedClass builder) {
                         constructor.param(field.type(), field.name())
                                 .annotate(annotation.getAnnotationClass())
                                 .param("value", str);
-                        body.assign(JExpr._this().ref(field.name()),
-                                // TODO(micah): this should respect immutability config
-                                immutableCopy(assignment, field.type()));
+
+                        if (this.getGenerationConfig().isImmutable()) {
+                            body.assign(JExpr._this().ref(field.name()),
+                                    immutableCopy(assignment, field.type()));
+                        } else {
+                            body.assign(JExpr._this().ref(field.name()), assignment);
```

Comment:
Ah, I was copying the control flow we've used in examining `isImmutable()` elsewhere in the plugin (the `*Rule.java` classes, for example). Should I change those to match as well?